### PR TITLE
Saurabh.ajmera/ravtdbq 353 cor variable issue inside projection and filter

### PR DESCRIPTION
**Pull request by sauajmera**
$cor Variable handling for Project and Filter clause in case of alias clause.

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 1]
```diff
@@ -2694,4 +2664,38 @@ private String getTableName(String alias, RelNode rel) {
...
+          List<RexNode> operands = ((RexCall) condition).operands;
+          int index = operands.get(0) instanceof RexSubQuery ? 0
+              : (operands.size() == 2 ? (operands.get(1) instanceof RexSubQuery ? 1 : -1) : -1);
+          op = index >= 0
+              ? ((RexSubQuery) (((RexCall) condition).operands.get(index))).op : null;
+        }
+      }
+      return correlOperators.contains(op);
+    } else if (rel instanceof LogicalProject && !rel.getVariablesSet().isEmpty()) {
+      return ((LogicalProject) rel).getChildExps()
```
Comment (Mahesh-Raut) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java:2694]: Please try to add a test case for your code changes.

**Comment by Mahesh-Raut** [comment]
Changes look good. L1 is done.

**Review by Mahesh-Raut — APPROVED** [review]
Changes look good. L1 is done.

**Comment by vijayjogi-dm** [comment]
L2 is done

**Pull request merged**